staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -240,14 +237,17 @@ func clearSecurityContext(p *corev1.Pod, containerName string) {
 }
 
 // disallowRoot configures the container to run as a non-root user.
+// If the container is already configured to run as a non-0 UID, that will be
+// retained; otherwise the UID will be updated to 65534.
 func disallowRoot(p *corev1.Pod, containerName string) {
 	podutils.VisitContainers(&p.Spec, podutils.AllContainers, func(c *corev1.Container, _ podutils.ContainerType) bool {
 		if c.Name != containerName {
 			return true
 		}
-		c.SecurityContext = &corev1.SecurityContext{
-			RunAsNonRoot: pointer.BoolPtr(true),
+		if c.SecurityContext == nil {
```

Comment:
We stop clearing SecurityContext for node, because we need to keep them explicitly?

Reply:
That's right, we don't want to lose the effects of having set the running as non-root & drop all capabilities. Instead we should clear SecurityContext prior to us setting the run as non-root & setting the drop-all-capabilities.